jwst/extern/configobj/validate.py

```diff
@@ -582,6 +585,8 @@ def __init__(self, functions=None):
             'pass': self._pass,
             'option': is_option,
             'force_list': force_list,
+            'is_datamodel' : is_datamodel,
+            'is_string_or_datamodel': is_string_or_datamodel,
```

Comment:
I think we want to avoid editing any code in `extern/configobj/` as it is vendorized code, i.e. we're just packaging the version used by `astropy`.  I think these function definitions can go in `jwst.stpipe.config_parser`, and the additions to the validator here:

https://github.com/spacetelescope/jwst/blob/64dc036a73422a135802dd2452fb706b9e12d8c6/jwst/stpipe/config_parser.py#L212-L215